src/metabase/driver/sql/query_processor.clj

```diff
@@ -1295,17 +1303,39 @@
         expr
         [:lower expr]))))
 
+(mu/defn ^:private maybe-cast-uuid-for-equality
+  "Comparing UUID fields against text values requires casting."
+  [driver field arg]
+  (if (and (isa? (:base-type (get field 2)) :type/UUID)
```

Comment:
You should  be looking at `:effective-type` in preference to `:base-type` if it's present